src/Features/Onboarding.php

```diff
@@ -731,19 +731,24 @@ public static function get_allowed_themes() {
 	}
 
 	/**
-	 * Let the app know that we will be showing the onboarding route, so wp-admin elements should be hidden while loading.
+	 * Set the admin full screen class when loading to prevent flashes of unstyled content.
 	 *
-	 * @param bool $is_loading Indicates if the `woocommerce-admin-is-loading` should be appended or not.
-	 * @return bool
+	 * @param bool $classes Body classes.
+	 * @return array
 	 */
-	public function is_loading( $is_loading ) {
-		$show_profiler = self::should_show_profiler();
```

Comment:
Is there a specific reason you didn't use this helper function anymore?

The slight flash does still happen when the onboarding wizard hasn't been completed yet and it auto redirects from **WooCommerce > Home**, not sure if this is also preventable?
I tried making use of the `self::should_show_profiler();` locally again but that didn't make a difference there.

Reply:
`should_show_profiler()` will return `true` even on other pages besides the homepage and profiler.

I think it's original meaning has been lost over iterations and needs updating semantically, but I'll save that for another PR.